src/app/application/services/data-source/index.ts

```diff
@@ -1,36 +1,97 @@
-import CallableEventSource, { isClosed } from './CallableEventSource'
+import CallableEventSource from './CallableEventSource'
 import type { Creator, Destroyer } from './DataSourcePool'
+export type { DataSourceResponse } from './DataSourcePool'
+
+type Configuration = {
+  interval?: number
+  retry?: (e: any) => Promise<any> | undefined
+}
+type RetryingEventSource = CallableEventSource<Configuration>
+type Hideable = EventTarget & { hidden: boolean }
+
+export const getSource = (doc: Hideable) => {
+  return (cb: (source: RetryingEventSource) => Promise<unknown>, config: Configuration = {}) => {
+    let attempts = 0
+    let iterations = 0
+    return new CallableEventSource<Configuration>(async function * (this: RetryingEventSource) {
+      const self = this
```

Comment:
**Question**: Why do we need to assign `this` to a local variable here? It doesn’t look like `self` is accessed in a different function body so this shouldn’t be necessary.

Reply:
Seems to be due to https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-callback-literal.md, but there's also a little more `self` usage elsewhere in related code.

I don't think this one is a biggie to be honest but I kinda don't mind either way. But I definitely don't want to be fiddling with removing eslint node rules from browser code in this PR.

FYI: I don't think we should be using this eslint rule here, but I think we should just leave as is at least for the moment.

Reply:
Since this lint rule clearly doesn’t apply to our project, we should rather just disable it and replace `self` with `this` here.

Reply:
Yes most probably, but not now. Feel free to make an issue.

Reply:
I'm gonna resolve this one, lmk if you need to reopen.